### PR TITLE
added helper functions for checking vectors

### PR DESCRIPTION
**Pull request by lornat75**

### CODE REVIEW

File: app/iCubTest/test_left_arm.ini [hunk 1]
```diff
@@ -5,7 +5,7 @@ portname /${robotname}/left_arm
 joints 16
 
 target  -30.0 25.0 10.0 40.0 -10.0 -10.0 10.0 50 15.0 15.0 15.0 15.0 15.0 15.0 15.0 15.0
-min      -3.0 -3.0 -3.0 -3.0 -3.0 -3.0 -3.0 -3.0 -3.0 -3.0 -3.0 -3.0 -3.0 -3.0 -3.0 -3.0
+min       3.0  3.0  3.0  3.0  3.0  3.0  3.0  3.0  3.0  3.0  3.0  3.0  3.0  3.0  3.0  3.0
```
Comment (traversaro) [inline app/iCubTest/test_left_arm.ini]: Spurious ?

Comment (lornat75) [inline app/iCubTest/test_left_arm.ini]: uhm, no. I thought it would be better to specify unsigned thresholds. Does it make sense?

Comment (traversaro) [inline app/iCubTest/test_left_arm.ini]: Ahhh ok got it.

Comment (traversaro) [inline app/iCubTest/test_left_arm.ini]: By the way, for this kind of function I always saw that in practice is very difficult to require an asymmetric tollerance .

**Pull request merged**